internal/zinc-core/src/main/scala/sbt/internal/inc/Stamp.scala

```diff
@@ -58,10 +59,10 @@ trait WithPattern { protected def Pattern: Regex }
 import java.lang.{ Long => BoxedLong }
 
 /** Define the hash of the file contents. It's a typical stamp for compilation sources. */
-final class Hash(val value: Array[Byte]) extends StampBase {
-  val hexHash = IOHash.toHex(value)
+final class Hash private[inc] (val hexHash: String) extends StampBase {
```

Comment:
`private[inc]` is no guarantee - as you see `Stamp.fromString` invokes it without doing any validation:

```scala
scala> sbt.internal.inc.Stamp.fromString("hash(gobbledygook)")
res0: xsbti.compile.analysis.Stamp = hash(gobbledygook)
```